src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,417 @@
+PARAM_MATRIX_OUT("score", "If test data is specified, this "
+    "matrix is where the class score for the test set will be saved.",
+    "p");
+
+size_t NumberOfClasses(const size_t numClasses,
+                       const arma::Row<size_t>& labels);
+
+static void mlpackMain()
+{
+  // Collect command-line options.
+  const double lambda = CLI::GetParam<double>("lambda");
+  const double delta = CLI::GetParam<double>("delta");
+  const bool intercept = CLI::HasParam("no_intercept") ? true : false;
+  const string optimizerType = CLI::GetParam<string>("optimizer");
+  const double tolerance = CLI::GetParam<double>("tolerance");
+  const size_t maxIterations = (size_t) CLI::GetParam<int>("max_iterations");
+
+  // One of training and input_model must be specified.
+  RequireAtLeastOnePassed({ "training", "input_model" }, true);
+
+  // If no output file is given, the user should know that the model will not be
+  // saved, but only if a model is being trained.
+  RequireAtLeastOnePassed({ "output_model", "predictions", "score"},
+      false, "no output will be saved");
+
+  ReportIgnoredParam({{ "test", false }}, "predictions");
+  ReportIgnoredParam({{ "test", false }}, "score");
+  ReportIgnoredParam({{ "test", false }}, "test_labels");
+
+  // Max Iterations needs to be positive.
+  RequireParamValue<int>("max_iterations", [](int x) { return x >= 0; },
+      true, "max_iterations must be positive or zero");
+
+  // Tolerance needs to be positive.
+  RequireParamValue<double>("tolerance", [](double x) { return x >= 0.0; },
+      true, "tolerance must be positive or zero");
+
+  // Optimizer has to be L-BFGS or ParallelSGD.
+  #ifdef HAS_OPENMP
+  RequireParamInSet<string>("optimizer", { "lbfgs", "psgd" },
+      true, "unknown optimizer");
+
+  if (optimizerType != "psgd")
+  {
+    if (CLI::HasParam("step_size"))
+    {
+      Log::Warn << PRINT_PARAM_STRING("step_size") << " ignored because "
+          << "optimizer type is not 'psgd'." << std::endl;
+    }
+    if (CLI::HasParam("shuffle"))
+    {
+      Log::Warn << PRINT_PARAM_STRING("shuffle") << " ignored because "
+          << "optimizer type is not 'psgd'." << std::endl;
+    }
+  }
+
+  // Step Size must be positive.
+  RequireParamValue<double>("step_size", [](double x) { return x >= 0.0; },
+      true, "step size must be positive");
+
+  #endif
+
+  // Lambda must be positive.
+  RequireParamValue<double>("lambda", [](double x) { return x >= 0.0; },
+      true, "lambda must be positive or zero");
+
+  // Number of Classes must be Non-Negative
+  RequireParamValue<int>("number_of_classes", [](int x) { return x >= 0; },
+                         true, "number of classes must be greater than or "
+                         "equal to 0 (equal to 0 in case of unspecified.)");
+
+  // Delta must be positive.
+  RequireParamValue<double>("delta", [](double x) { return x >= 0.0; }, true,
+      "Margin of difference between correct class and other classes");
```

Comment:
I think we need an actual error here, maybe adding ` must be nonnegative` to the existing message is sufficient.